engine/execution/rpc/engine.go

```diff
@@ -802,6 +806,53 @@ func (h *handler) blockHeaderResponse(header *flow.Header) (*execution.BlockHead
 	}, nil
 }
 
+// GetTransactionExecutionMetricsAfter gets the execution metrics for a transaction after a given block.
+func (h *handler) GetTransactionExecutionMetricsAfter(
+	_ context.Context,
+	req *execution.GetTransactionExecutionMetricsAfterRequest,
+) (*execution.GetTransactionExecutionMetricsAfterResponse, error) {
+	height := req.GetBlockHeight()
+
+	metrics, err := h.transactionMetrics.GetTransactionExecutionMetricsAfter(height)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get metrics after block height %v: %v", height, err)
+	}
+
+	response := &execution.GetTransactionExecutionMetricsAfterResponse{
+		Results: make([]*execution.GetTransactionExecutionMetricsAfterResponse_Result, 0, len(metrics)),
+	}
+
+	for blockHeight, blockMetrics := range metrics {
```

Comment:
So far there was no guarantee given that this would be sorted. Do you think it would be nice to add that?

Reply:
I feel like it would be reasonable for a client to assume that it is sorted by height since all of the other endpoints work that way. you could document in the protobuf that it's not if you don't want to make the guarantee, but it's pretty simple to add and this will be a low volume endpoint